src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,304 +1,500 @@
+    else:
+        raise ValueError('Accepted formats: FASTQ or BAM\n'
+                         'Could not determine {} file type from URL:\n{}'.format(uuid, url))
+
+    rm_secondary = job.wrapJobFn(run_samtools_view,
+                                 get_bam.rv(),
+                                 flag='0x800',
+                                 ncores=config.cores,
+                                 disk=config.file_size, cores=config.cores)
+
+    sort_bam_disk = PromisedRequirement(lambda x: 3 * x.size + human2bytes('10G'),
+                                        rm_secondary.rv())
+    sort_bam = job.wrapJobFn(run_samtools_sort,
+                             rm_secondary.rv(),
+                             ncores=config.cores,
+                             cores=config.cores,
+                             disk=sort_bam_disk)
+
+    index_bam_disk = PromisedRequirement(lambda x: int(1.5 * x.size), sort_bam.rv())
+    index_bam = job.wrapJobFn(run_samtools_index, sort_bam.rv(), disk=index_bam_disk)
+
+    job.addChild(get_bam)
+    get_bam.addChild(rm_secondary)
+    rm_secondary.addChild(sort_bam)
+    sort_bam.addChild(index_bam)
+    return sort_bam.rv(), index_bam.rv()
+
+
+def setup_and_run_bwa_kit(job, url, config):
+    """
+    Downloads and aligns paired FASTQs to reference genome using BWA.
+
+    :param JobFunctionWrapping Job: passed by Toil
+    :param str url: FASTQ sample URL
+    :param Namespace config: Input parameters and shared FileStoreIDs
+    :return: BAM FileStoreID
+    :rtype: str
+    """
+    job.fileStore.logToMaster("Aligning sample: {}".format(config.uuid))
+    cores = multiprocessing.cpu_count()
+    # Download fastq files
+    fq1 = job.addChildJobFn(download_url_job,
+                            url,
+                            name='toil.1.fq',
+                            synapse_login=config.synapse_login,
+                            s3_key_path=config.ssec,
+                            disk=config.file_size)
+    # Assumes second fastq url is identical
+    fq2_url = url.replace('1.fq', '2.fq')
+    fq2 = job.addChildJobFn(download_url_job,
+                            fq2_url,
+                            name='toil.2.fq',
+                            synapse_login=config.synapse_login,
+                            s3_key_path=config.ssec,
+                            disk=config.file_size)
+
+    bwa_config = deepcopy(config)
+    bwa_config.r1 = fq1.rv()
+    bwa_config.r2 = fq2.rv()
+    # bwa_alignment uses a different naming convention
+    bwa_config.ref = config.genome_fasta
+    bwa_config.fai = config.genome_fai
+    bwakit_disk = PromisedRequirement(lambda x, y: 3 * (x.size + y.size) + human2bytes('8G'),
+                                      fq1.rv(), fq2.rv())
+    return job.addFollowOnJobFn(run_bwakit,
+                                bwa_config,
+                                cores,
+                                sort=False,
```

Comment:
I thought it would be simpler to have one point in the pipeline where all samples get sorted.